### ISSUE TITLE
Run 'await this.ShowMessageAsync()' in BackgroundWorker

### ISSUE THREAD

**Issue by akhalid** [MahApps/MahApps.Metro#1935]
Hello. Is there any way to run the ShowMessageAsync() in a BackgroundWorker thread? Thanks.

**Comment by jammerxd** [comment]
Yes there is, I use an invoke method:

Application.Dispatcher.Invoke(aysnc()=>{
    await this.ShowMessageAsync("TITLE","MESSAGE");
});

**Comment by akhalid** [comment]
Hi. Thanks for the response. I am trying to use it in the worker_DoWork() function but is not working. I had to change to:

Application.Current.Dispatcher.Invoke(aysnc()=>{
                        await this.ShowMessageAsync("TITLE","MESSAGE");
                        });
because I got the error:
An object reference is required for the non-static field, method, or property 'System.Windows.Threading.DispatcherObject.Dispatcher.get'

However now I get the error: 
The name 'async' does not exist in the current context

Not sure what to do, any ideas? Thanks!
